packages/cdk-assets/lib/private/handlers/container-images.ts

```diff
@@ -68,6 +52,36 @@ export class ContainerImageAssetHandler implements IAssetHandler {
       file: source.dockerFile,
     });
   }
+
+  protected async getImageUri(ecr: AWS.ECR, destination: DockerImageDestination): Promise<string | null> {
+    const account = (await this.host.aws.discoverCurrentAccount()).accountId;
+    const repoUri = await repositoryUri(ecr, destination.repositoryName);
+
+    if (!repoUri) {
+      throw new Error(`No ECR repository named '${destination.repositoryName}' in account ${account}. Is this account bootstrapped?`);
+    }
+
+    const imageUri = `${repoUri}:${destination.imageTag}`;
+
+    this.host.emitMessage(EventType.CHECK, `Check ${imageUri}`);
+    if (await imageExists(ecr, destination.repositoryName, destination.imageTag)) {
+      this.host.emitMessage(EventType.FOUND, `Found ${imageUri}`);
+      return null;
+    }
+
+    if (this.host.aborted) { return null; }
```

Comment:
I see you're trying to collapse some cases here for the calling code, but I feel this is mixing concerns. It's a little confusing that this code's `null` return value means: "the image already exists OR the progress host asked us to stop".

It's more logical to me that a `null` returned here means: "the image already exists OR I don't know because I didn't check" (already a *little* dubious but I'm willing to go with it), and that the caller STILL checks for `if (this.host.aborted || imageUri === null) { return; }`.

Yes some cases collapse to the same `null` but the reasons for those values are more clearly in line with the responsibility of the method, and the code will be less likely to break because there are non-obvious relationships between parts of the code.